In [1]:
from scipy.stats import t
from math import sqrt

In [2]:
n1 = 25
x1_mean = 3.6 
n2 = 27
x2_mean = 3.8

In [3]:
sigma2_1 = 0.36
sigma2_2 = 0.32

Let's assume that the samples are independant. In this case we can use the two-sample t-test with one-sided alternative

In [7]:
t_val = (x2_mean-x1_mean) / sqrt(sigma2_1/n1 + sigma2_2/n2)

In [8]:
t_val

1.2343832596089048

In [17]:
t.ppf(0.95, df=n1+n2-2)

1.6759050245283311

As we assume to check whether the second distribution's mean is significantly greater than the first one's we would use a one-sided test.

So we need to check where the 95-th percentile of t-distribution with n1+n2-2 degrees of freedom is less than the t-value for our samples.

The critical area in our case looks the following way: $(1,6759, +\infty)$. Our t-value $t_{val} = 1.234$ is not in the critical area so we cannot reject the null hypothesis that the samples are taken from the same distribution.

In [16]:
t.cdf(t_val, df=n1+n2-2)

0.8885848594177619